.github/workflows/stream_core_flutter_workflow.yml

```diff
@@ -0,0 +1,71 @@
+name: stream_core_flutter_workflow
+
+env:
+  FLUTTER_CHANNEL: stable
+  FLUTTER_VERSION: 3.29.0
+  ENV_PROPERTIES: ${{ secrets.ENV_PROPERTIES }}
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  analyze:
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          cache: true
+          channel: ${{ env.FLUTTER_CHANNEL }}
+          flutter-version: ${{ env.FLUTTER_VERSION }}
```

Comment:
Let's remove the flutter-version so that it can use the latest one. Also I think only setting `cache: true` will not cache it properly, I faced this earlier in our repo's.

We need to set a cache key along with it to work correctly.

```yaml
          cache: true
          cache-key: flutter-:os:-:channel:-:version:-:arch:-:hash:-${{ hashFiles('**/pubspec.lock') }}
          ```